src/staking-v3/hooks/useAprV3.ts

```diff
@@ -0,0 +1,103 @@
+import { $api } from 'boot/api';
+import { ethers } from 'ethers';
+import { ref, watch } from 'vue';
+import { EraLengths } from '../logic';
+import { useDappStaking } from './useDappStaking';
+
+interface InflationParam {
+  maxInflationRate: string;
+  adjustableStakersPart: string;
+  baseStakersPart: string;
+  idealStakingRate: string;
+  bonusPart: string;
+}
+
+export const useAprV3 = () => {
+  const stakerApr = ref<number>(0);
+  const bonusApr = ref<number>(0);
+  const { eraLengths, isVotingPeriod } = useDappStaking();
+
+  const percentageToNumber = (percent: string): number => {
+    return parseFloat(String(percent)) * 0.01;
+  };
+
+  const toAstr = (wei: string): number => {
+    return Number(ethers.utils.formatEther(BigInt(wei)));
+  };
+
+  const getCyclePerYear = (eraLength: EraLengths): number => {
+    const secBlockProductionRate = 12;
+    const secsOneYear = 365 * 24 * 60 * 60;
+    const periodLength =
+      eraLength.standardErasPerBuildAndEarnPeriod + eraLength.standardErasPerVotingPeriod;
+
+    // Todo: fetch it via API
+    const periodsPerCycle = 2;
+    const eraPerCycle = periodLength * periodsPerCycle;
+    const blocksStandardEraLength = eraLength.standardEraLength;
+    const blockPerCycle = blocksStandardEraLength * eraPerCycle;
+    const cyclePerYear = secsOneYear / secBlockProductionRate / blockPerCycle;
+    return cyclePerYear;
+  };
+
+  const getApr = async (): Promise<{ stakerApr: number; bonusApr: number }> => {
+    try {
+      const apiRef = $api!;
+      const eraLengthRef = eraLengths.value;
+      if (!eraLengthRef.standardEraLength) {
+        return { stakerApr: 0, bonusApr: 0 };
+      }
+
+      const [inflation, currentEraInfo, totalIssuanceRaw] = await Promise.all([
+        apiRef.query.inflation.inflationParams(),
+        apiRef.query.dappStaking.currentEraInfo(),
+        apiRef.query.balances.totalIssuance(),
+      ]);
+
+      const inflationParams = inflation.toHuman() as unknown as InflationParam;
```

Comment:
This can be dangerous in case if runtime model changes. The code will silently fail without compilation errors. Do something like https://github.com/AstarNetwork/astar-apps/blob/9576d94a4a2ccccb6296dcc68e49c3edeeb95075/src/staking-v3/logic/repositories/DappStakingRepository.ts#L456

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/1113/commits/eb7b59c9d4d18718e7be2e209bedcf449439ff26) commit.